### PR TITLE
Modal for teachers of at risk gated students

### PR DESCRIPTION
**Pull request by juanmanzojr**
Adds the At Risk Age Gated Students Modal and Table Components and related components as well. The table makes use of the currreent ManageStudentsRedux and filters the students who are at risk of being age gated and displays their child account compliance state.

Screenshot 2024-05-30 at 5 12 47 PM

## Links

- jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/P20/boards/69?assignee=712020%3Af0e2716f-09f9-4686-8490-9f1f9a08bb06&selectedIssue=P20-817)

## Testing story
Added new Unit tests for the Modal itself and the Table.

## Follow-up work
Will add UI testing and eyes test once the banner is merged.

### CODE REVIEW

**Comment by wilkie** [comment]
Could you talk a little more about any kind of testing you have done to investigate scale? Have you stress tested this with an account with a lot of sections / students?

File: apps/src/templates/manageStudents/AgeGatedTableConsentStatusCell.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+class AgeGatedTableConsentStatusCell extends Component<Props> {
+  static propTypes = {
+    id: PropTypes.number.isRequired,
+    consentStatus: PropTypes.string.isRequired,
+  };
+
+  getConsentStatus = () => {
+    switch (this.props.consentStatus) {
+      case 'l':
```
Comment (artem-vavilov) [inline apps/src/templates/manageStudents/AgeGatedTableConsentStatusCell.tsx]: Please use values from shared constants: https://github.com/code-dot-org/code-dot-org/blob/08eefbc8ae14f750d3fb6b38d8d656081d011344/lib/cdo/shared_constants.rb#L651-L657

**Review by daynew — CHANGES_REQUESTED** [review]
Make sure to include the relevant Amplitude metrics https://codedotorg.atlassian.net/browse/P20-904.

 Example: https://github.com/code-dot-org/code-dot-org/blob/11b5e11e9a43112135b4a591b0b26eac57dd3e9a/apps/src/templates/policy_compliance/ParentalPermissionBanner/index.tsx#L33

File: apps/i18n/common/en_us.json [hunk 2]
```diff
@@ -410,6 +410,14 @@
   "childAccountConsentValidPermission": "Permission:",
   "childAccountConsentValidPermissionGranted": "Granted on {date}",
   "childAccountConsentValidMessage": "Thank you for approving your child’s Code.org account! You will receive a confirmation email within 24–48 hours. Please let your child know you’ve approved their account so they can get started learning and coding today!",
+  "childAccountPolicy_consentStatus": "Consent Status",
+  "childAccountPolicy_consentProcessReadMore": "Read more about this [consent process and these changes here]({url}).",
+  "childAccountPolicy_lockedOut": "Locked Out",
+  "childAccountPolicy_studentParentalConsentStatus": "Student Parental Consent Status",
+  "childAccountPolicy_studentParentalConsentNotice": "The following students will require parental consent by 7/1/2024 to continue using Code.org with their personal account. Please ensure these students provide a parent or guardian email address to complete the parental consent process.",
```
Comment (daynew) [inline apps/i18n/common/en_us.json:417]: Normally the "7/1/2024" should be a dynamic value pulled from the CPA configuration, but I think this string will change soon due to the Grace Period feature.

File: apps/src/lib/util/AnalyticsConstants.js [hunk 3]
```diff
@@ -47,6 +47,7 @@ const EVENTS = {
   CAP_PARENT_EMAIL_MODAL_UPDATED: 'CAP Parent Email Modal Updated',
   CAP_PARENT_EMAIL_MODAL_RESEND: 'CAP Parent Email Modal Resend',
   CAP_PARENT_EMAIL_MODAL_CLOSED: 'CAP Parent Email Modal Closed',
+  CAP_AGE_GATED_MODAL_SHOWN: 'CAP Age Gated Modal Shown',
```
Comment (daynew) [inline apps/src/lib/util/AnalyticsConstants.js]: I think there area few metrics missing: 
1. Banner Shown (this was missing from the JIRA originally).
2. Button in banner clicked.
3. Documentation link clicked.

I realize the button being clicked and the modal would be redundant, but lets do this for now just incase we change behavior in the future (such as auto showing the modal).

I have updated the [JIRA](https://codedotorg.atlassian.net/browse/P20-904) task to only include the work being done in this PR and added the metric.

Also could you also make sure the metric event name matches what is in the JIRA? I have updated them to follow the pattern in this code.

**Review by daynew — CHANGES_REQUESTED** [review]
Just some metrics which need to be added.

File: apps/src/lib/util/AnalyticsConstants.js (hunk 3 above)
Comment (juanmanzojr) [reply apps/src/lib/util/AnalyticsConstants.js]: I plan on adding the Banner metrics in the following [PR](https://github.com/code-dot-org/code-dot-org/pull/58481).

File: apps/src/templates/policy_compliance/AgeGatedStudentsModal/AgeGatedStudentsModal.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        data-testid="age-gated-students-modal"
+      >
+        <div>
+          <h2 className={styles.modalHeader}>
+            {i18n.childAccountPolicy_studentParentalConsentStatus()}
+          </h2>
+          <hr />
+          <p>{i18n.childAccountPolicy_studentParentalConsentNotice()}</p>
+          <br />
+          <a href="#" onClick={modalDocumentationClicked}>
```
Comment (daynew) [inline apps/src/templates/policy_compliance/AgeGatedStudentsModal/AgeGatedStudentsModal.tsx:70]: I created a [JIRA](https://codedotorg.atlassian.net/browse/P20-962) to figure out the actual value for this.

**Review by daynew — APPROVED** [review]

File: apps/src/templates/policy_compliance/AgeGatedStudentsModal/AgeGatedStudentsModal.tsx (hunk 4 above)
Comment (juanmanzojr) [reply apps/src/templates/policy_compliance/AgeGatedStudentsModal/AgeGatedStudentsModal.tsx:70]: Thanks! Logic is being add in the banner PR as well: https://github.com/code-dot-org/code-dot-org/pull/58481

**Pull request merged**